crates/sui/src/client_commands.rs

```diff
@@ -732,6 +732,17 @@ impl SuiClientCommands {
                 serialize_signed_transaction,
                 lint,
             } => {
+                if build_config.test_mode {
```

Comment:
While the solution presented in this PR works, a user running `sui client publish --help` will still see `--test` in the list of available flags.

In order to remove this entirely, I can see two solutions:
1. [`BuildConfig`](https://github.com/MystenLabs/sui/blob/main/external-crates/move/tools/move-package/src/lib.rs#L36) is duplicated into a `struct` without a `test_mode` field, and used here
2. The `test_mode` flag is extracted into a different structure `BuildConfigNoTest` that two fields: it, and athe original `BuildConfig`

Both feel like overengineering, so I chose this approach instead.

Reply:
This makes sense to me as well -- agreed that the other options feel a bit like overengineering as well.